src/pymor/reductors/aaa.py

```diff
@@ -0,0 +1,411 @@
+    data or a given full-order model which can be a |TransferFunction| or any model which
+    has a `transfer_function` attribute. MIMO and non-parametric data is accepted. See
+    :cite:`NST18` for the non-parametric and :cite:`CRBG20` for the parametric version of
+    the algorithm.
+
+    Parameters
+    ----------
+    sampling_values
+        Values where sample data has been evaluated or the full-order model should be evaluated.
+        Sampling values are represented as a nested list `svs` such that `svs[i]` corresponds
+        to sampling values of the `i-th` variable. For |TransferFunctions| or models with a
+        `transfer_function` attribute, the first variable corresponds to frequency data.
+    samples_or_fom
+        Can be either a full-order model (|TransferFunction| or |Model| with a `transfer_function`
+        attribute) or data sampled at the values specified in `sampling_values`. Samples are
+        represented as a tensor `S`. E.g., for 3 inputs `S[i,j,k]` corresponds to the sampled
+        value at `(sampling_values[0][i],sampling_values[1][j],sampling_values[2][k])`. In the
+        MIMO case `S[i,j,k]` represents a matrix of dimension `dim_output` times `dim_input`.
+    conjugate
+        Whether to compute complex conjugates of first sampling variables and enforce
+        interpolation in complex conjugate pairs (allows for constructing real system matrices).
+    nsp_tol
+        Tolerance for null space of higher-dimensional Loewner matrix to check for
+        interpolation or convergence.
+    post_process
+        Whether to do post-processing or not. If the Loewner matrix has a null space
+        of dimension greater than 1, it is assumed that the algorithm converged to
+        a non-minimal order interpolant which may cause numerical issues. In this case,
+        the post-processing procedure computes an interpolant of minimal order.
+    L_rk_tol
+        Tolerance for ranks of 1-D Loewner matrices computed in post-processing.
+    """
+
+    def __init__(self, sampling_values, samples_or_fom, conjugate=True, nsp_tol=1e-16, post_process=True,
+                 L_rk_tol=1e-8):
+        if isinstance(samples_or_fom, TransferFunction) or hasattr(samples_or_fom, 'transfer_function'):
+            fom = samples_or_fom
+            if not isinstance(samples_or_fom, TransferFunction):
+                fom = fom.transfer_function
+            self.num_vars = 1 + fom.parameters.dim
+
+            assert len(sampling_values) == self.num_vars
+            self.parameters = fom.parameters
+            self.samples = np.empty([len(sv) for sv in sampling_values] + [fom.dim_output, fom.dim_input],
+                                    dtype=sampling_values[0].dtype)
+            for idx, vals in zip(np.ndindex(self.samples.shape[:-2]),
+                                 itertools.product(*sampling_values)):
+                params = {p: v for p, v in zip(fom.parameters, vals[1:])}
```

Comment:
I guess this should be extended to include higher-dimensional parameters. I guess `params = fom.parameters.parse(vals[1:])` should work.